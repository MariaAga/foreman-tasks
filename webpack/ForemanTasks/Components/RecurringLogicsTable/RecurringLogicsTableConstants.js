export const RECURRING_LOGICS = 'RECURRING_LOGICS';
