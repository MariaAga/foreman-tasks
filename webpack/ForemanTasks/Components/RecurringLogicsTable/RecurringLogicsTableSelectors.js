import { createSelector } from 'reselect';
import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';

import {
  getLastOccurrence,
  getNextOccurrence,
} from './RecurringLogicsTableHelper';
import { RECURRING_LOGICS } from './RecurringLogicsTableConstants';

const selectResponse = state => selectAPIResponse(state, RECURRING_LOGICS);

export const selectResults = createSelector(selectResponse, ({ results }) =>
  results
    ? results.map(result => ({
        cronLine: result.cron_line,
        taskCount: result.tasks.length,
        action: result.tasks[0] ? result.tasks[0].action : '-',
        lastOccurrence: getLastOccurrence(result.tasks),
        nextOccurrence: getNextOccurrence(result.state, result.tasks),
        currentIteration: result.iteration,
        iterationLimit: result.max_iteration,
        repeatUntil: result.end_time,
        state: result.state,
        id: result.id,
        canEdit: result.can_edit,
        TaskGroupId: result.task_group_id,
      }))
    : []
);
