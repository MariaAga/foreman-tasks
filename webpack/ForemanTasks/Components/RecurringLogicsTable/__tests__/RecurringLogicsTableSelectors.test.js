import { STATUS } from 'foremanReact/constants';
import { RECURRING_LOGICS } from '../RecurringLogicsTableConstants';
import { selectResults } from '../RecurringLogicsTableSelectors';

const recurringLogics = {
  max_iteration: 3,
  id: 63,
  cron_line: '51 17 * * *',
  end_time: null,
  iteration: 1,
  state: 'cancelled',
  can_edit: true,
  task_group_id: 123,
  tasks: [
    {
      start_at: '2020-02-24 17:51:00 +0200',

      action: 'Run hosts job: Run n',
      started_at: null,
    },
  ],
};
const logicsState = {
  API: {
    [RECURRING_LOGICS]: {
      response: { results: [recurringLogics] },
      status: STATUS.PENDING,
    },
  },
};

describe('RecurringLogicsTableSelectors', () => {
  it('selectResults', () => {
    expect(selectResults(logicsState)).toMatchSnapshot();
  });
});
