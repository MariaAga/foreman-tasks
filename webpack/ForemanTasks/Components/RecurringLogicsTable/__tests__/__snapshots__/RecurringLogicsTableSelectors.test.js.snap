// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurringLogicsTableSelectors selectResults 1`] = `
Array [
  Object {
    "TaskGroupId": 123,
    "action": "Run hosts job: Run n",
    "canEdit": true,
    "cronLine": "51 17 * * *",
    "currentIteration": 1,
    "id": 63,
    "iterationLimit": 3,
    "lastOccurrence": "-",
    "nextOccurrence": "-",
    "repeatUntil": null,
    "state": "cancelled",
    "taskCount": 1,
  },
]
`;
